### DISCUSSION TITLE
Rendering occurs on the useController.

### DISCUSSION THREAD

**Discussion by yoe88** [react-hook-form/react-hook-form#11835] [Q&A]
Hello

I'm trying both Controller and useController.
https://github.com/orgs/react-hook-form/discussions/7852
Looking at this document, I think they are the same function.
However, it seems that rendering does not occur where the Controller is used, but it seems that rendering occurs where the useController is used. (Please check the console window.)
Is it correct that rendering occurs when using the useController?
If so, is there a way that renderings do not occur even when using the useController?

Here's an example.
https://codesandbox.io/p/sandbox/react-hook-form-controller-h72x5q?file=%2Fsrc%2FApp.js%3A12%2C6

**Comment by xeeds** [comment]
I think it is as easy as there is no state change in the component using Controller, hence no rerender. Try the following :

```
import { Controller, useFormContext } from "react-hook-form";
import InputLength from "./InputLength";
import { useState } from "react";

function ControllerDiv() {
  const [someState, setSomeState] = useState(false);
  console.log("ControllerDiv render");
  const { control } = useFormContext();

  function onChange(e, field) {
    const v = e.target.value.trim();
    field.onChange(v);
  }

  return (
    <div style={{ border: "1px solid black", padding: "5px" }}>
      <h1>Controller</h1>
      <Controller
        control={control}
        name="controller"
        rules={{ required: "required controller value" }}
        render={({ field }) => (
          <InputLength
            {...field}
            onChange={(e) => {
              onChange(e, field);
              setSomeState(!someState);
            }}
            maxLength={15}
          />
        )}
      />
    </div>
  );
}

export default ControllerDiv;
```
